### PR TITLE
Fix/investigate session finding user logic

### PR DESCRIPTION
**Pull request by szymonkadas**
# Summary: 
- User with deleted account can't see nor edit other users profile data. Also user can now login to create account (clicking on createProfile also automatically creates account due to github authorization)
- finding user/profiles by user id's/email are now safekeeped from falsy values. where clause with falsy value returned first value from db. 
- profile pages and components are now also looking at session.user.id (probably) because of src/lib/auth.ts authOptions.callbacks jwt function that returned object with id null. Object that is indeed defined, so hence not falsy.

# Notes:
Problem has probably occurred mainly due to "if" conditions not enough strict for actual data behaviour, and tricky prisma findFirst functionality (undefined values would prove to unexpected behaviours).

### CODE REVIEW

**Review by wawrzonnn — APPROVED** [review]

**Review by mmcode9407 — APPROVED** [review]
Looks good! :)

File: src/backend/user/user.service.ts [hunk 1]
```diff
@@ -18,6 +18,10 @@ export async function getUsersPayload() {
 }
 
 export async function getUserById(id: string) {
+  if (!id) {
```
Comment (Sumick) [inline src/backend/user/user.service.ts:21]: **nitpick:** We should not need that, as `id` is required

File: src/backend/user/user.service.ts [hunk 2]
```diff
@@ -37,6 +41,10 @@ export async function getUserById(id: string) {
 }
 
 export async function findUserByEmail(email: string) {
+  if (!email) {
+    return null
```
Comment (Sumick) [inline src/backend/user/user.service.ts:45]: **nitpick:** We should not need that, as `email` is required

File: src/components/CreateProfileBtn/CreateProfileBtn.tsx [hunk 3]
```diff
@@ -16,7 +15,7 @@ const CreateProfileBtn = () => {
   }
 
   const onClickHandler = async () => {
-    if (!session) {
+    if (!session?.user?.id) {
```
Comment (Sumick) [inline src/components/CreateProfileBtn/CreateProfileBtn.tsx:18]: **nitpick:** We should leave as it is, and make session null if it's not present

File: src/components/Headers/AppHeader/AppHeader.tsx [hunk 4]
```diff
@@ -19,7 +18,7 @@ const AppHeader = async () => {
     ? await getProfileByUserEmail(session.user.email)
     : null
 
-  if (session) {
+  if (session?.user?.id) {
```
Comment (Sumick) [inline src/components/Headers/AppHeader/AppHeader.tsx:21]: **nitpick:** We should leave as it is, and make session null if it's not present

**Review by Sumick — CHANGES_REQUESTED** [review]
Instead of changing so many things here, we should be able to make it from one place. Maybe just let's return null instead of an object with id of null? Then you would not have to change other things and add more checks
Zrzut ekranu 2023-12-19 o 15 38 54

**Comment by szymonkadas** [comment]
Closing due to too many conflicts from changes from the meantime.

**Pull request closed without merging**